docs/en/sql-reference/functions/time-series-functions.md

```diff
@@ -0,0 +1,48 @@
+---
+slug: /en/sql-reference/functions/time-series-functions
+sidebar_position: 172
+sidebar_label: Time Series
+---
+
+# Time Series Functions
+
+Below functions are used for time series analysis.
+
+## seriesPeriodDetectFFT
+
+Finds the period of the given time series data using FFT
+Detect Period in time series data using FFT.
+FFT - Fast Fourier transform (https://en.wikipedia.org/wiki/Fast_Fourier_transform)
+
+**Syntax**
+
+``` sql
+seriesPeriodDetectFFT(series);
+```
+
+**Arguments**
+
+- `series` - An array of numeric values
+
+**Returned value**
+
+- A real value equal to the period of time series
+- Returns NAN when number of data points are less than four.
+
+Type: [Float64](../../sql-reference/data-types/float.md).
+
+**Examples**
+
+Query:
+
+``` sql
+SELECT seriesPeriodDetectFFT([1, 4, 6, 1, 4, 6, 1, 4, 6, 1, 4, 6, 1, 4, 6, 1, 4, 6, 1, 4, 6]) AS print_0;
```

Comment:
Could you please add examples with periodic functions?

E.g. sinusoid or https://en.wikipedia.org/wiki/Triangle_wave:

```
SELECT seriesPeriodDetectFFT(arrayMap(x -> (abs(((x - 1) % 4) - 2) - 1), range(1000)))

Query id: 6911842f-1d14-4688-9b4f-adfc79f5ea19

┌─seriesPeriodDetectFFT(arrayMap(lambda(tuple(x), minus(abs(minus(modulo(minus(x, 1), 4), 2)), 1)), range(1000)))─┐
│                                                                                                               4 │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

```

Side question: How would it perform on asymptotic functions (e.g. `tan(x)`)?

Reply:
Added more test-cases (with triangle and square waves) and update examples too. I also compared the method with python for asymptotic functions: https://pastila.nl/?028974ec/f182f872d708823af2d1b832180ff663#krxGEW8PRVF+I5R5/N+uwQ==
The results are similar. Also, FFT is suited towards detecting period in a time series data.